app/src/main/java/com/swent/skillswap/ui/editUser/EditUserScreen.kt

```diff
@@ -0,0 +1,355 @@
+/**
+ * @author Léonard MARTI 394185 /!\ Written with help of Copilot /!\
+ * > Helped me finding the right compose functions, complete all the repetitive code (construction
+ * > of instances for example)
+ */
+package com.swent.skillswap.ui.editUser
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.outlined.ArrowBack
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.filled.Person
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.AsyncImage
+import com.swent.skillswap.ui.theme.*
+import com.swent.skillswap.ui.utils.*
+import com.swent.skillswap.viewModel.EditUserViewModel
+
+object EditUserTags {
+    const val GO_BACK_BUTTON = "edit_user_go_back_button"
+    const val USERNAME_TEXTFIELD = "edit_user_username_textfield"
+    const val EMAIL_TEXTFIELD = "edit_user_email_textfield"
+    const val VALIDATE_BUTTON = "edit_user_validate_button"
+    const val PROFILE_PICTURE = "edit_user_profile_picture"
+    const val SKILLSET_SECTION = "edit_user_skillset_section"
+    const val GENERAL_ERROR = "edit_user_general_error"
+    const val SUCCESS_MESSAGE = "edit_user_success_message"
+}
+
+/** Displays the edit user screen. */
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
+@Composable
+fun EditUserScreen(
+    vm: EditUserViewModel = viewModel(),
+    onGoBack: () -> Unit = {},
+) {
+    val uiState by vm.uiState.collectAsState()
+    val user = uiState.editedUser
+
+    var username by remember { mutableStateOf(user?.username ?: "") }
+    var email by remember { mutableStateOf(user?.email ?: "") }
```

Comment:
When uiState.editedUser changes, the remembered username and email will not automatically update, because remember only runs once per composition. You should use 

```
LaunchedEffect(user?.username, user?.email) {
    username = user?.username ?: ""
    email = user?.email ?: ""
}
```